x/group/module/module.go

```diff
@@ -155,6 +161,42 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 	return []abci.ValidatorUpdate{}
 }
 
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(
+			provideModuleBasic,
+			provideModule,
+		),
+	)
+}
+
+func provideModuleBasic() runtime.AppModuleBasicWrapper {
+	return runtime.WrapAppModuleBasic(AppModuleBasic{})
+}
+
+type groupInputs struct {
+	depinject.In
+
+	Config           *modulev1.Module
+	Key              *store.KVStoreKey
+	Cdc              codec.Codec
+	AccountKeeper    group.AccountKeeper `key:"cosmos.auth.v1.AccountKeeper"`
+	BankKeeper       group.BankKeeper    `key:"cosmos.bank.v1.Keeper"`
+	Registry         cdctypes.InterfaceRegistry
+	MsgServiceRouter *baseapp.MsgServiceRouter
+}
+
+func provideModule(in groupInputs) (keeper.Keeper, runtime.AppModuleWrapper) {
+	/*
+		Example of setting group params:
+		in.Config.MaxMetadataLen = 1000
+	*/
+	k := keeper.NewKeeper(in.Key, in.Cdc, in.MsgServiceRouter, in.AccountKeeper, group.Config{MaxExecutionPeriod: in.Config.MaxExecutionPeriod.AsDuration(), MaxMetadataLen: in.Config.MaxMetadataLen})
```

Comment:
so now we have 2 ways of defining the group config, through proto generated `*modulev1.Module` and through our existing `Config` regular golang struct (x/group/config.go)
I'm wondering if we could now avoid this duplication cc/ @aaronc

Reply:
Makes sense. I think it's maybe okay for now...